pipeline.py

```diff
@@ -1,43 +1,54 @@
 from prefect import task, Flow
 
 import pandas as pd
+import os
 import fsspec
+import zarr
 
 
-url_pattern = 'https://www.ncei.noaa.gov/data/sea-surface-temperature-optimum-interpolation/access/avhrr-only/{{ %Y%m }}/avhrr-only-v2.{{ %Y%m%d }}.nc'
-
-date_range = {'start': '1981-09-01', 'end': '2019-08-11', 'freq': 'D'}
-dates = pd.date_range(**date_range)
-files = dates.strftime(url_pattern)
-
-
+def src_url_from_key(key):
+    source_url_pattern = 'https://www.ncei.noaa.gov/data/sea-surface-temperature-optimum-interpolation/v2.1/access/avhrr/{yyyymm}/oisst-avhrr-v02r01.{yyyymmdd}.nc'
+    return source_url_pattern.format(yyyymm=key.strftime('%Y%m'), yyyymmdd=key.strftime('%Y%m%d'))
+    
+cache_location = 'gs://pangeo-scratch/rabernat/pangeo_smithy/oisst-avhrr-v02r01-cache/'
+    
+    
 @task
-def download(source):
-    import wget
-
-    wget.download(source)
-
-    return target
+def download(key):
+    source_url = src_url_from_key(key)
+    os.path.join(cache_location, str(hash(key)))
+    target_url = os.path.join(cache_location, str(hash(key)))
+    
+    # there is probably a better way to do caching!
+    try:
+        fsspec.open(target_url).open()
+        return target_url
+    except FileNotFoundError:
+        pass
+    
+    with fsspec.open(source_url, mode='rb') as source:
+        with fsspec.open(target_url, mode='wb') as target:
+            target.write(source.read())
+    return target_url
 
 
 @task
-def combine(sources):
+def combine_and_publish(sources, target):
```

Comment:
I combined these into one step because I'm not sure that xarray is smart enough to be able to serialize the h5netcdf-backed arrays which point a fsspec files. Easier to just do these operations in a single task.